e2e/davinci-app/components/combobox.ts

```diff
@@ -0,0 +1,227 @@
+import type { MultiSelectCollector, Updater } from '@forgerock/davinci-client/types';
+
+/**
+ * Creates a dropdown component that allows selecting multiple options
+ * from a predefined list
+ * @param {HTMLFormElement} formEl - The form element to attach the multi-select to
+ * @param {SingleSelectCollector} collector - Contains the options and configuration
+ * @param {Updater} updater - Function to call when selection changes
+ */
+export default function multiSelectDropdownComponent(
+  formEl: HTMLFormElement,
+  collector: MultiSelectCollector,
+  updater: Updater,
+) {
+  // Create a container for the multi-select
+  const containerDiv = document.createElement('div');
+  containerDiv.className = 'multi-select-container';
+
+  // Create a label for the multi-select
+  const label = document.createElement('label');
+  label.textContent = collector.output.label || 'Select options';
+  label.className = 'multi-select-label';
+  containerDiv.appendChild(label);
+
+  // Create the main dropdown button
+  const dropdownButton = document.createElement('button');
+  dropdownButton.type = 'button';
+  dropdownButton.className = 'multi-select-button';
+  dropdownButton.textContent = 'Select options';
+
+  // Create the dropdown container
+  const dropdown = document.createElement('div');
+  dropdown.className = 'multi-select-dropdown';
+  dropdown.style.display = 'none';
+
+  // Store selected values
+  const selectedValues: string[] = [];
+
+  // Function to update selected options display
+  const updateSelectionDisplay = () => {
+    if (selectedValues.length === 0) {
+      dropdownButton.textContent = 'Select options';
+    } else {
+      // Get labels for selected values
+      const selectedLabels = collector.output.options
+        .filter((option) => selectedValues.includes(option.value))
+        .map((option) => option.label);
+
+      if (selectedLabels.length <= 2) {
+        dropdownButton.textContent = selectedLabels.join(', ');
+      } else {
+        dropdownButton.textContent = `${selectedLabels.length} options selected`;
+      }
+    }
+  };
+
+  // Function to update the updater with current selection
+  const updateSelection = () => {
+    // If single value, send as string, otherwise send as array
+    if (selectedValues.length === 1) {
+      updater(selectedValues[0]);
+    } else {
+      // FIXED: Pass the array directly instead of joining with commas
+      updater(selectedValues);
+    }
+  };
+
+  // Populate dropdown with options
+  collector.output.options.forEach((option) => {
```

Comment:
Rather than all of this, I'd just recommend using the native `<select multiple>` option: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/multiple.